src/transformers/training_args.py

```diff
@@ -1771,6 +1771,13 @@ def __post_init__(self):
                     self.accelerator_config = AcceleratorConfig()
                 elif isinstance(self.accelerator_config, dict):
                     self.accelerator_config = AcceleratorConfig(**self.accelerator_config)
+                # Check that a user didn't pass in the class instantiator
+                # such as `accelerator_config = AcceleratorConfig`
+                elif callable(self.accelerator_config):
```

Comment:
interesting that this falls in the `isinstance` of `AcceleratorConfig` 🤣

Reply:
Because when we’re in this situation, whatever is passed is of type `type`!

Reply:
While `callable` is more encompassing, it may be good to just do `type`